coral-common/src/main/java/com/linkedin/coral/common/catalog/IcebergHiveTableConverter.java

```diff
@@ -0,0 +1,114 @@
+ * and {@link com.linkedin.coral.hive.hive2rel.functions.HiveFunctionResolver} infrastructure expects a
+ * Hive {@code org.apache.hadoop.hive.metastore.api.Table} object for:
+ * <ul>
+ *   <li>Dali UDF resolution (extracting "functions" and "dependencies" from table properties)</li>
+ *   <li>Table identification (database name, table name)</li>
+ *   <li>Ownership and permission checks (owner field)</li>
+ * </ul>
+ *
+ * <p>Rather than refactoring the entire call chain to accept {@link Dataset},
+ * this converter provides a pragmatic bridge that allows Iceberg tables to work seamlessly with the existing
+ * Hive-based infrastructure.
+ *
+ * <p><b>What gets converted:</b>
+ * <ul>
+ *   <li>Iceberg schema → Hive columns (via {@code HiveSchemaUtil.convert()})</li>
+ *   <li>All Iceberg table properties → Hive table parameters (including Dali UDF metadata)</li>
+ *   <li>Table metadata (name, owner, timestamps, table type)</li>
+ *   <li>Storage descriptor with SerDe info (for compatibility)</li>
+ * </ul>
+ */
+public class IcebergHiveTableConverter {
+
+  private IcebergHiveTableConverter() {
+    // Utility class - prevent instantiation
+  }
+
+  /**
+   * Converts IcebergDataset to a Hive Table object for backward compatibility with function resolution.
+   *
+   * @param icebergDataset Iceberg dataset to convert
+   * @return Hive Table object with complete metadata and schema
+   */
+  public static Table toHiveTable(IcebergDataset icebergDataset) {
+    org.apache.iceberg.Table icebergTable = icebergDataset.getIcebergTable();
+
+    // Parse db.table name (format: "dbname.tablename")
+    String fullName = icebergDataset.name();
+    String dbName;
+    String tableName;
+    int dotIndex = fullName.indexOf('.');
+    if (dotIndex > 0) {
+      dbName = fullName.substring(0, dotIndex);
+      tableName = fullName.substring(dotIndex + 1);
+    } else {
+      // Fallback if no dot (shouldn't happen in practice)
+      dbName = "default";
+      tableName = fullName;
+    }
+
+    // Convert Iceberg schema to Hive columns using HiveSchemaUtil
+    StorageDescriptor storageDescriptor = new StorageDescriptor();
+    SerDeInfo serDeInfo = new SerDeInfo();
+
+    // Copy all Iceberg table properties to Hive table parameters
+    // This includes Dali UDF metadata ("functions", "dependencies") and any other custom properties
+    Map<String, String> hiveParameters = new HashMap<>(icebergDataset.properties());
+
+    // Set SerDe parameters (include avro.schema.literal if present)
+    Map<String, String> serdeParams = new HashMap<>();
+    if (hiveParameters.containsKey("avro.schema.literal")) {
+      serdeParams.put("avro.schema.literal", hiveParameters.get("avro.schema.literal"));
+    }
+    serDeInfo.setParameters(serdeParams);
+    storageDescriptor.setSerdeInfo(serDeInfo);
+
+    // Convert Iceberg schema to Hive columns
+    try {
+      storageDescriptor.setCols(HiveSchemaUtil.convert(icebergTable.schema()));
```

Comment:
Any particular reason why we choose to set storageDescriptor columns from HiveSchemaUtil.convert(icebergTable.schema()) and not from AvroSchemaUtil.convert(hiveParameters.get("avro.schema.literal"))?